model-api/src/commonMain/kotlin/org/modelix/model/api/INode.kt

```diff
@@ -47,6 +47,15 @@ interface INode {
 
     fun tryGetConcept(): IConcept? = getConceptReference()?.tryResolve()
 
+    /**
+     * Replaces this node with a new node of the given concept that uses the same id as this node.
+     * Properties, references and children will be the same.
+     *
+     * @param concept the concept of the new node
+     * @return replacement for this node with the new given concept
+     */
+    fun replaceNode(concept: ConceptReference): INode
```

Comment:
```suggestion
    fun replaceNode(concept: ConceptReference): INode { throw UnsupportedOperationException() }
```
Default implementation required to avoid a breaking change.

Reply:
No longer breaking since it's been moved to a new interface.